specs/phase1/beacon-chain.md

```diff
@@ -1057,51 +536,124 @@ def process_epoch(state: BeaconState) -> None:
     process_reveal_deadlines(state)
     process_challenge_deadlines(state)
     process_slashings(state)
+    process_pending_headers(state)
+    charge_confirmed_header_fees(state)
+    reset_pending_headers(state)
     process_final_updates(state)  # phase 0 final updates
     process_phase_1_final_updates(state)
 ```
 
+#### Pending headers
+
+```python
+
+def process_pending_headers(state: BeaconState):
+    for slot in range(SLOTS_PER_EPOCH):
+        for shard in range(SHARD_COUNT):
+            # Pending headers for this (slot, shard) combo
+            candidates = [
+                c for c in state.previous_epoch_pending_shard_headers if
+                (c.slot, c.shard) == (slot, shard)
+            ]
+            if True not in [c.confirmed for c in candidates]:
+                # The entire committee (and its balance)
+                full_committee = get_beacon_committee(state, slot, shard)
```

Comment:
`full_committee` and `full_committee_balance` computations can be outside of the loop